### ISSUE TITLE
Insecure password handling

### ISSUE THREAD

**Issue by benjick** [jwalton/gh-ecr-push#3]
Thanks for making this action!

I get the following warning when I run it:

```
WARNING! Using --password via the CLI is insecure. Use --password-stdin.
WARNING! Your password will be stored unencrypted in /home/runner/.docker/config.json.
```

**Comment by jwalton** [comment]
Basically, what `docker login` is trying to warn you about in the first case is that if you do:

```sh
$ docker login --username me --password supersecretpassword
```

Then you've probably just written your docker password to your bash history.  That's bad, because if someone else gets access to your machine, they can just read your bash password off the disk.

The second warning is similar; docker by default writes your password somewhere unencrypted, which isn't terrible secure.  There's some fancy ways around by using the [credentials store](https://docs.docker.com/engine/reference/commandline/login/#credentials-store) to store your passwords securely.

The question is, can these insecurities be exploited in your Github Actions VM?

Obviously the Github staff have access to the VM, and could read your password out of it...  But, they also have access to the raw secrets you have in your repo, so they don't need to do anything too fancy to steal your docker password.

It's not impossible that some other third party Github action or a malicious dependency could try to read this data off your VM and send it somewhere - for example if you're a node.js developer, and you use the popular Mocha testing framework, and Mocha decides to go rogue, they could read that data and send it somewhere (assuming you are running tests after pushing to docker, that is).  But, I'm not sure most of the credential managers would save you here - if you're using `pass` to store credentials, for example, a malicious script can just as easily run `pass` and fetch your credentials from there.

A possible solution to the first warning would be, instead of running:

```sh
$ docker login --username me --password supersecretpassword
```

we could run:

```sh
$ docker login --username me
```

and pipe the password in via stdin, keeping it out of the bash history.

A possible solution to the second problem is to simply delete /home/runner/.docker/config.json after doing the push, so it's unavailable for anything that runs after the push command.  This would make it *more* secure, but someone who is targeting your build specifically could probably recover the contents of the deleted file, if they were really working at it.

**Comment by benjick** [comment]
Thanks, but I'm not doing any login. Just using your action.

**Comment by jwalton** [comment]
Sorry - I should have been more clear.  The action is doing the login.  :)

**Comment by benjick** [comment]
Oh, I see, sorry 🙏 Would you open to a PR?

**Comment by jwalton** [comment]
For sure.

**Comment by jwalton** [comment]
Fixed by #11, published in v1.3.0.
